integrations/gitlab/gitlab_integration/utils.py

```diff
@@ -10,6 +11,23 @@
 RETRY_TRANSIENT_ERRORS = True
 
 
+def generate_gitlab_client(host: str, token: str) -> Gitlab:
+    try:
+        gitlab_client = Gitlab(
+            host, token, retry_transient_errors=RETRY_TRANSIENT_ERRORS
+        )
+        gitlab_client.auth()
+        logger.info("Successfully authenticated using the provided private token")
+    except gitlab.exceptions.GitlabAuthenticationError:
+        gitlab_client = Gitlab(
+            host, oauth_token=token, retry_transient_errors=RETRY_TRANSIENT_ERRORS
+        )
+        gitlab_client.auth()
+        logger.info("Successfully authenticated using the provided OAuth2.0 token")
```

Comment:
is there some pattern in the token which we can evaluate whether to connect with oauth or regular token?